ledger/src/bank_forks.rs

```diff
@@ -27,24 +28,15 @@ pub struct SnapshotConfig {
     pub snapshot_path: PathBuf,
 }
 
-#[derive(Debug)]
+#[derive(Error, Debug)]
 pub enum BankForksError {
-    SnapshotError(SnapshotError),
-    SnapshotPackageSendError(SnapshotPackageSendError),
-}
-pub type Result<T> = std::result::Result<T, BankForksError>;
+    #[error("snapshot error")]
+    SnapshotError(#[from] SnapshotError),
 
-impl std::convert::From<SnapshotError> for BankForksError {
-    fn from(e: SnapshotError) -> BankForksError {
-        BankForksError::SnapshotError(e)
-    }
-}
-
-impl std::convert::From<SnapshotPackageSendError> for BankForksError {
-    fn from(e: SnapshotPackageSendError) -> BankForksError {
-        BankForksError::SnapshotPackageSendError(e)
-    }
+    #[error("snapshot package send error")]
+    SnapshotPackageSendError(#[from] SnapshotPackageSendError),
 }
+pub type Result<T> = std::result::Result<T, BankForksError>;
```

Comment:
We probably shouldn't `pub` these `Result` aliases. It's generally better when downstream code defines their own error objects.